### PR TITLE
feat: add destination_ranges and source_ranges in firewall rules

### LINKED ISSUE #461: Firewall rules for terraform-google-network module
### TL;DR

Hi there, one of our enterprise customers currently uses terraform-google-modules/terraform-google-network in their IaC pipeline, and they would like to have source and destination to be used in egress and ingress firewall rules. 

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

just FYI: Below is the comment from our customer: 
***
Seems like it doesn’t support ingress rule with destination range, neither egress rule with source range.
 
https://github.com/terraform-google-modules/terraform-google-network/blob/v7.0.0/modules/firewall-rules/main.tf#L24
  source_ranges           = each.value.direction == "INGRESS" ? each.value.ranges : null
  destination_ranges      = each.value.direction == "EGRESS" ? each.value.ranges : null
***

### PR DESCRIPTION
**Pull request by imrannayer**
fix #461

### CODE REVIEW

File: modules/firewall-rules/main.tf [hunk 1]
```diff
@@ -14,8 +14,14 @@
  * limitations under the License.
  */
 
+locals {
+  ingress_rules = { for rules in var.ingress_rules : rules.name => merge(rules, { direction = "INGRESS" }) }
```
Comment (g-awmalik) [inline modules/firewall-rules/main.tf:18]: nit:

```suggestion
  ingress_rules = { for rule in var.ingress_rules : rule.name => merge(rule, { direction = "INGRESS" }) }
```

File: modules/firewall-rules/main.tf [hunk 2]
```diff
@@ -14,8 +14,14 @@
...
+  egress_rules  = { for rules in var.egress_rules : rules.name => merge(rules, { direction = "EGRESS" }) }
```
Comment (g-awmalik) [inline modules/firewall-rules/main.tf:19]: nit: as I was reading it i got confused for a second why all "rules" when it's being applied to each rule.

```suggestion
  egress_rules  = { for rule in var.egress_rules : rule.name => merge(rule, { direction = "EGRESS" }) }
```

File: examples/firewall-rules/main.tf
Comment (g-awmalik) [inline examples/firewall-rules/main.tf:1]: Can we call this example "bidirectional-firewall-rules"? Something to indicate that we're creating both ingress and egress rules here.

File: examples/firewall-rules/versions.tf
Comment (g-awmalik) [inline examples/firewall-rules/versions.tf:1]: Don't think you need versions.tf in the example.

**Review by g-awmalik — APPROVED** [review]
Thanks for updating this. LGTM except for a few nits.

File: examples/firewall-rules/versions.tf
Comment (imrannayer) [reply examples/firewall-rules/versions.tf:1]: versions.tf is in every example of this and other modules. Are we planning to remove them from examples?

**Pull request merged**